packages/babel-parser/src/parser/expression.js

```diff
@@ -1488,7 +1489,11 @@ export default class ExpressionParser extends LValParser {
         first = false;
       } else {
         this.expect(tt.comma);
-        if (this.eat(tt.braceR)) break;
+        if (this.match(tt.braceR)) {
+          this.addExtra(node, "trailingComma", this.state.lastTokStart);
```

Comment:
What if we instead record the `traillingCommaAttachedNodeStart`? So we can always throw at the node.

Reply:
I'd prefer to make `trailingComma` a boolean, and then throw at `props[props.length - 1].start`